x-pack/plugins/security_solution/common/search_strategy/security_solution/index.ts

```diff
@@ -218,6 +219,76 @@ export type StrategyResponseType<T extends FactoryQueryTypes> = T extends HostsQ
   ? UsersRelatedHostsStrategyResponse
   : never;
 
+export type StrategyRequestInputType<T extends FactoryQueryTypes> = T extends HostsQueries.hosts
```

Comment:
A neat trick we can use here to avoid maintaining this large conditional is the TypeScript `Extract` feature. Since we have all our request schemas in one discriminated union `searchStrategyRequestSchema`, we can create this generic type based on that union:

```
type SearchStrategyRequestInputSchema = z.input<typeof searchStrategyRequestSchema>;

export type RequestSchemaType<T extends FactoryQueryTypes> = Extract<SearchStrategyRequestInputSchema, {factoryQueryType: T}>;

export type Test = RequestSchemaType<HostsQueries.hosts>;
```
This code makes the `Test` type equivalent to `StrategyRequestInputType<HostsQueries.hosts>`. We can do the same thing for the existing code (`StrategyRequestType` and `StrategyResponseType`) that established this pattern and reduce some of the tedious code maintenance here.